### ISSUE TITLE
imap_getmailboxes returns false;

### ISSUE THREAD

**Issue by ursulnegrul** [ddeboer/imap#134]
In Connection.php:149 

`
   if (null === $this->mailboxNames) {
            $mailboxes = imap_getmailboxes($this->resource, $this->server, '*');
`

$mailboxes is false, when var_dumping the $connection i get 

`object(Ddeboer\Imap\Connection)[4]
  private 'server' => string '{::FFFF:127.8.0.1:993/imap/notls/ssl/user="example@gmail.com"}' (length=61)
  private 'resource' => resource(7, imap)
  private 'mailboxes' => null
  private 'mailboxNames' => null
`
I assumed the 'server' is probably incorrectly set, so by manually replacing 'server' with 
`{imap.gmail.com:993/imap/ssl}`

$mailboxes is set correctly.

I am running on Windows localhost, PHP7.

**Comment by Slamdunk** [comment]
Hi, the error message you provide shows an IPv6, but if you replacd it with the second server it should be ok. Maybe the issue is related to https://github.com/ddeboer/imap/issues/167 : can you check it please?

**Comment by robertodormepoco** [comment]
Please fix this, it is in plain sight, just check [the imap-ext source code in php-src](https://github.com/php/php-src/blob/8d3f8ca12a0b00f2a74a27424790222536235502/ext/imap/php_imap.c#L1924), as you can see the [global structure](https://github.com/php/php-src/blob/8d3f8ca12a0b00f2a74a27424790222536235502/ext/imap/php_imap.c#L814) tested is null most of the times, so there are no guarantees about it being an array, you can also dig inside the c-client imap library (sorry I'm not able to find a versioned one here on GitHub, any apt source will do).

- mail.c:913 calls the DRIVER (a generic interface) list method and as an example in the imap4rc1.c concrete instance you can follow the imap_list_work method, that via the imap_send
```
IMAPPARSEDREPLY *imap_send (MAILSTREAM *stream,char *cmd,IMAPARG *args[])
```
will populate the global reply structure if it has any response

or just read the the php doc for mail_list which has almost the same logic as the aforementioned one

```
 Returns an array containing the names of the mailboxes or false in case of failure. 
```

**Comment by Slamdunk** [comment]
Closed with #372
